app/scripts/modules/kubernetes/src/v2/manifest/selector/selector.component.ts

```diff
@@ -0,0 +1,66 @@
+import { IComponentOptions, IQService, IController, module } from 'angular';
+
+import { AccountService, IAccountDetails } from '@spinnaker/core';
+import { KUBERNETES_MANIFEST_LABEL_EDITOR } from './labelEditor/labelEditor.component';
+import { IManifestSelector } from './IManifestSelector';
+
+class KubernetesManifestSelectorCtrl implements IController {
+  public selector: IManifestSelector;
+  public accounts: IAccountDetails;
+  public kindsMetadata: string;
+
+  constructor(private $q: IQService, private accountService: AccountService) {
+    'ngInject';
+    const dataToFetch = {
```

Comment:
Would only this pattern if you're planning on making other async calls in the future in this controller. If not, this can just be
```
this.accountService.getAllAccountDetailsForProvider('kubernetes', v2').then(accounts => {
  this.accounts = accounts;
});
```